### PR TITLE
Enable logits extraction from vLLM for training

### PR DESCRIPTION
**Pull request by MooMoo-Yang**
This PR adds support for directly extracting logits from vLLM during training.
By enabling this feature, we avoid redundant computation of sequence logits during reinforcement learning (RL) fine-tuning, which previously required separate forward passes.

The change introduces a new flag:

```
--use_vllm_logprobs
```

When enabled, the training engine will use logits returned by vLLM instead of computing them again inside the trainer.

### CODE REVIEW

**Comment by MooMoo-Yang** [comment]
To evaluate the effectiveness of this change, we conducted four comparative experiments:

1. **Synchronous training** using the original training engine's logits.
2. **Asynchronous training** using the original training engine's logits.
3. Training using **vLLM-provided logits with BF16 precision**.
4. Training using **vLLM-provided logits with FP32 precision**, by modifying the model's `lm_head` to FP32 (to reduce numerical errors).

---

### ✅ Experimental Results
comparison



We found that using vLLM's **FP32 logits** achieves training performance comparable to that of the original engine, although a small performance gap still exists.

**Comment by hijkzzz** [comment]
The async LLM for the agentic RL also needs to support vLLM logits.

File: examples/scripts/train_reinforce_baseline_llama_ray_async_vllm_logits.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
+set -x
```
Comment (hijkzzz) [inline examples/scripts/train_reinforce_baseline_llama_ray_async_vllm_logits.sh]: remove this script~~

**Comment by MooMoo-Yang** [comment]
> The async LLM for the agentic RL also needs to support vLLM logits.

Updated the corresponding training code. Local debugging shows no errors currently. Please review when you have time.

**Comment by hijkzzz** [comment]
It needs to wait until the FP32 lm_head MR of vLLM is merged.
